File: NextSpring/src/main/java/com/study/nextspring/global/jpa/entity/BaseEntity.java
package com.study.nextspring.global.jpa.entity;

import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.MappedSuperclass;
import lombok.AccessLevel;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;

import static jakarta.persistence.GenerationType.IDENTITY;

@Getter
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
@MappedSuperclass
public class BaseEntity {
    @Id
    @GeneratedValue(strategy = IDENTITY) // AUTO_INCREMENT
    @Setter(AccessLevel.PRIVATE)
    @EqualsAndHashCode.Include
    private Long id;
}

File: k_rstudy/src/main/kotlin/com/krstudy/kapi/global/jpa/BaseEntity.kt
package com.krstudy.kapi.global.jpa

import jakarta.persistence.EntityListeners
import jakarta.persistence.MappedSuperclass
import lombok.Getter
import lombok.ToString
import org.springframework.data.annotation.CreatedDate
import org.springframework.data.annotation.LastModifiedDate
import org.springframework.data.jpa.domain.support.AuditingEntityListener
import java.time.LocalDateTime


@MappedSuperclass
@Getter
@EntityListeners(AuditingEntityListener::class)
@ToString(callSuper = true)
open class BaseEntity : IdEntity() {
    @CreatedDate
    private var createDate: LocalDateTime? = null

    @LastModifiedDate
    private var modifyDate: LocalDateTime? = null

    // Getters and Setters
    fun getCreateDate(): LocalDateTime? = createDate
    fun setCreateDate(createDate: LocalDateTime?) { this.createDate = createDate }

    fun getModifyDate(): LocalDateTime? = modifyDate
    fun setModifyDate(modifyDate: LocalDateTime?) { this.modifyDate = modifyDate }
}
